pymc3/distributions/continuous.py

```diff
@@ -4257,3 +4273,179 @@ def logcdf(self, value):
+    Support   :math:`x \in (0, \infty)`
+    Mean      :math:`dfrac{h}{4} if :math:`z=0`, :math:`\dfrac{tanh(z/2)h}{2z}` otherwise.
+    Variance  :math:`0.041666688h` if :math:`z=0`, :math:`\dfrac{h(sinh(z) - z)(1 - tanh^2(z/2))}{4z^3}` otherwise.
+    ========  =============================
+
+    Parameters
+    ----------
+    h: float, optional
+        The shape parameter of the distribution (h > 0).
+    z: float, optional
+        The exponential tilting parameter of the distribution.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        rng = np.random.default_rng()
+        with pm.Model():
+            x = pm.PolyaGamma('x', h=1, z=5.5)
+        with pm.Model():
+            x = pm.PolyaGamma('x', h=25, z=-2.3, rng=rng, size=(100, 5))
+
+    References
+    ----------
+    .. [1] Polson, Nicholas G., James G. Scott, and Jesse Windle.
+           "Bayesian inference for logistic models using Pólya–Gamma latent
+           variables." Journal of the American statistical Association
+           108.504 (2013): 1339-1349.
+    .. [2] Windle, Jesse, Nicholas G. Polson, and James G. Scott.
+           "Sampling Polya-Gamma random variates: alternate and approximate
+           techniques." arXiv preprint arXiv:1405.0506 (2014)
+    .. [3] Luc Devroye. "On exact simulation algorithms for some distributions
+           related to Jacobi theta functions." Statistics & Probability Letters,
+           Volume 79, Issue 21, (2009): 2251-2259.
+    .. [4] Windle, J. (2013). Forecasting high-dimensional, time-varying
+           variance-covariance matrices with high-frequency data and sampling
+           Pólya-Gamma random variates for posterior distributions derived
+           from logistic likelihoods.(PhD thesis). Retrieved from
+           http://hdl.handle.net/2152/21842
+    """
+    rv_op = polyagamma
+
+    @classmethod
+    def dist(cls, h=1.0, z=0.0, rng=None, size=None, **kwargs):
+        # really not sure whether h & z are required to be tensors? The
+        # distribution uses numpy arrays or scalars for parameter values.
+        hh = aet.as_tensor_variable(h)
+        zz = aet.as_tensor_variable(z)
+
+        msg = f"The variable {hh} specified for PolyaGamma has non-positive "
+        msg += "values, making it unsuitable for this parameter."
+        Assert(msg)(hh, aet.all(aet.gt(hh, 0.00001)))
+
+        return super().dist([h, z], size=size, rng=rng, **kwargs)
+
+    def logp(value, h, z):
+        """
+        Calculate log-probability of Polya-Gamma distribution at specified value.
+
+        Parameters
+        ----------
+        value: numeric
+            Value(s) for which log-probability is calculated. If the log
+            probabilities for multiple values are desired the values must be
+            provided in a numpy array.
+
+        Returns
+        -------
+        TensorVariable
+        """
+        return bound(polyagamma_logpdf(value, h, z), h > 0, value > 0)
```

Comment:
Does this `polyagamma_logpdf` function return an Aesara graph?  If not, this won't work.

If the computations in `polyagamma_logpdf` are too complicated to be represented symbolically in Aesara, then you'll need to make an `Op` that wraps this function.  That `Op` will also need to be capable of NumPy-like broadcasting.